File: algorithm/apple_stocks.swift
func getMaxProfit(from prices: [Int]) -> Int {
  guard prices.count >= 2 else { return 0 }
  
  var result = 0
  var buyValue = prices.first!
  
  for index in 1..<prices.count {
    
    let sellValue = prices[index]
    
    if sellValue < buyValue {
      
      buyValue = sellValue
    } else {
      
      let dealValue = sellValue - buyValue
      result = max(dealValue, result)
    }
  }
 
  return result
}

let prices = [10, 7, 5, 8, 4, 11, 9]
let prices2: [Int] = []
let prices3 = [10, 7]
print(getMaxProfit(from: prices3))

File: algorithm/apple_stocks.js
function getMaxProfit(arr) {
  
  if (arr.length === 0) { return 0 }
  
  let result = 0;
  let buy;
  
  for (var value of arr) {
    
    if (buy === undefined) {
      
      buy = value;
      console.log("buy is initialized to:", buy);
      continue;
    }
    
    if (value < buy) {
      
      buy = value
      console.log("buy is updated to", buy);
    } else {
      result = Math.max(value - buy, result);
      console.log("result is updated", result);
    }
    
  }
  
  return result;
}

let stock = [10, 7, 5, 8, 11, 9];

console.log("Result is",getMaxProfit(stock));